test/functional/test_framework/wallet.py

```diff
@@ -259,13 +260,17 @@ def create_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node=None, utx
             tx.wit.vtxinwit = [CTxInWitness()]
             tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE]), bytes([LEAF_VERSION_TAPSCRIPT]) + self._internal_key]
         tx_hex = tx.serialize().hex()
+        txid = tx.rehash()
+        wtxid = tx.getwtxid()
+
+        if mempool_valid is not None:
```

Comment:
I don't think we need the miniwallet to check that `mempool_valid=False` is equal to tmpa["allowed"].
```suggestion
        if mempool_valid:
```

Reply:
Good idea, done.